tests/tools/onert_train/src/onert_train.cc

```diff
@@ -180,38 +180,44 @@ int main(const int argc, char **argv)
       exit(-1);
     }
 
-    // NOTE: Measuring memory can't avoid taking overhead. Therefore, memory will be measured on the
-    // only warmup.
-    if (verbose == 0)
+    const int num_sample = data_length / tri.batch_size;
+    const int num_epoch = args.getEpoch();
+    for (uint32_t epoch = 0; epoch < num_epoch; ++epoch)
     {
-      phases.run(
-        "WARMUP",
-        [&](const benchmark::Phase &, uint32_t) { NNPR_ENSURE_STATUS(nnfw_run(session)); }, 1);
-      phases.run(
-        "EXECUTE",
-        [&](const benchmark::Phase &, uint32_t) { NNPR_ENSURE_STATUS(nnfw_run(session)); }, 1,
-        true);
-    }
-    else
-    {
-      phases.run(
-        "WARMUP",
-        [&](const benchmark::Phase &, uint32_t) { NNPR_ENSURE_STATUS(nnfw_run(session)); },
-        [&](const benchmark::Phase &phase, uint32_t nth) {
-          std::cout << "... "
-                    << "warmup " << nth + 1 << " takes " << phase.time[nth] / 1e3 << " ms"
-                    << std::endl;
-        },
-        1);
-      phases.run(
-        "EXECUTE",
-        [&](const benchmark::Phase &, uint32_t) { NNPR_ENSURE_STATUS(nnfw_run(session)); },
-        [&](const benchmark::Phase &phase, uint32_t nth) {
-          std::cout << "... "
-                    << "run " << nth + 1 << " takes " << phase.time[nth] / 1e3 << " ms"
-                    << std::endl;
-        },
-        1, true);
+      for (uint32_t n = 0; n < num_sample; ++n)
+      {
+        // get batchsize data
+        if (!generator(n, input_data, expected_data))
```

Comment:
@jyoungyun I have no idea what `generator` generate. I found

https://github.com/Samsung/ONE/blob/52430200a2aa3b79b8ed53e2eac70f003ccabaeb/tests/tools/onert_train/src/rawdataloader.h#L31-L33

But it provides no information. Could you please explain me what `generator` generate?